src/linagora.esn.unifiedinbox/views/unified-inbox/subheader.pug

```diff
@@ -17,5 +17,6 @@ block right
     .inbox-selection-aware-subheader.inbox-selecting(ng-switch-when="true")
       inbox-subheader-not-spam-button(ng-click="ctrl.unSpam()", ng-show="ctrl.canUnSpamMessages()")
       inbox-subheader-delete-button(ng-click="ctrl.moveToTrash()", ng-show="ctrl.canTrashMessages()")
-      inbox-subheader-mark-as-read-button(ng-click="ctrl.markAsRead()")
+      inbox-subheader-mark-as-read-button(ng-click="ctrl.markAsRead()", ng-if="ctrl.isUnreadItems() || ctrl.mixedItems()")
+      inbox-subheader-mark-as-unread-button(ng-click="ctrl.markAsUnread()", ng-if="ctrl.isReadItems() || ctrl.mixedItems()")
```

Comment:
I find it easier to:
* show the markAsRead button if isAtLeastOneUnreadItem
* show the markAsUnread button if isAtLeastOneReadItem

Reply:
easier what do you mean ?

Reply:
I mean easier to understand, and more optimized, simpler.
If you have to explain it you would say "The button to mark as read is here if at least on mail is unread" and "The button to mark as unread is here if at least on mail is read", and that makes sense in my opinion.
So yes it means removing the mixedItems and modifying and renaming the 2 other methods.

Reply:
the first one is easy to understand
I will keep it more simpler